github/search.go

```diff
@@ -54,6 +55,37 @@ func (s *SearchService) Repositories(query string, opt *SearchOptions) (*Reposit
 	return result, resp, err
 }
 
+// CommitsSearchResult represents the result of a commits search.
+type CommitsSearchResult struct {
+	Total             *int           `json:"total_count,omitempty"`
+	IncompleteResults *bool          `json:"incomplete_results,omitempty"`
+	Commits           []CommitResult `json:"items,omitempty"`
```

Comment:
I definitely think the pointer version is better despite any inconsistencies, especially for new structs. See #180 for discussion.

As for `Commits`, that's fine with me.